### PR TITLE
[CINFRA-187] Replicated Logs Maintenance Integration Tests

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose

Adding integration tests for replicated logs. This tests the complete stack of maintenance and replicated logs.

### CODE REVIEW

**Review by apetenchea — APPROVED** [review]
I think it looks good.

File: tests/js/server/replication2/replication2-maintenance-replicated-log-cluster.js [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    return function () {
+        let {current} = readReplicatedLogAgency(database, logId);
+        if (current === undefined) {
+            return Error("current not yet defined");
+        }
+        if (!current.leader) {
+            return Error("Leader has not yet established its term");
+        }
+        if (!current.leader.committedParticipantsConfig || current.leader.committedParticipantsConfig.generation < generation) {
+            return Error("Leader has not yet acked new generation");
```
[the remark this answers is not in the document]
Comment (maierlars) [reply tests/js/server/replication2/replication2-maintenance-replicated-log-cluster.js:78]: done

File: tests/js/server/replication2/replication2-maintenance-replicated-log-cluster.js [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+const replicatedLogIsReady = function (logId, term, participants, leader) {
+    return function () {
+        let {current} = readReplicatedLogAgency(database, logId);
+        if (current === undefined) {
+            return Error("current not yet defined");
+        }
+
+        for (const srv of participants) {
+            if (!current.localStatus || !current.localStatus[srv] || current.localStatus[srv].term < term) {
+                return Error(`Participant ${srv} has not yet acknowledge the current term.`);
```
[the remark this answers is not in the document]
Comment (maierlars) [reply tests/js/server/replication2/replication2-maintenance-replicated-log-cluster.js]: done

File: arangod/Replication2/ReplicatedLog/AgencyLogSpecification.cpp [hunk 3]
```diff
@@ -37,6 +37,10 @@ using namespace arangodb;
 using namespace arangodb::replication2;
 using namespace arangodb::replication2::agency;
 
+namespace {
+auto constexpr StringCommittedParticipantsConfig = std::string_view{"committedParticipantsConfig"};
```
[the remark this answers is not in the document]
Comment (maierlars) [reply arangod/Replication2/ReplicatedLog/AgencyLogSpecification.cpp]: does this rule apply to all constants or only for those part of an enum?

**Pull request merged**